react/src/components/ImportFromHuggingFaceModal.tsx

```diff
@@ -132,7 +158,41 @@ const ImportFromHuggingFaceModal: React.FC<ImportFromHuggingFaceModalProps> = ({
     formRef.current
       ?.validateFields()
       .then((values) => {
-        // TODO: Implement import from Hugging Face
+        importAndStartService.mutate(
+          {
+            huggingFaceUrl: values.url,
+            importOnly: isImportOnly,
+            serviceName: values.service_name || undefined,
+            folderName: values.folder_name || undefined,
+          },
+          {
+            onSuccess(data: any) {
+              if (data?.folder?.id && data?.folder?.name) {
+                upsertNotification({
+                  key: 'ImportFromHuggingFaceModal' + data.folder.id,
+                  message: `${t('data.modelStore.NewModelStoreFolderHasBeenCreated')}: ${data.folder.name}`,
+                  open: true,
+                  duration: 0,
+                  toText: t('data.modelStore.OpenFolder'),
+                  to: `/data?tab=model-store&folder=${data.folder.id}`,
+                });
+              }
+              if (data?.service?.endpoint_id && data?.service?.name) {
+                upsertNotification({
+                  key: 'ImportFromHuggingFaceModal' + data.service.endpoint_id,
+                  message: `${t('data.modelStore.NewServiceHasBeenCreated')}: ${data.service.name}`,
+                  open: true,
+                  duration: 0,
+                  toText: t('data.modelStore.ViewServiceInfo'),
+                  to: `/serving/${data.service.endpoint_id}`,
+                });
+              }
+            },
+            onError(e) {
+              message.error(e.message || t('dialog.ErrorOccurred'));
+            },
+          },
+        );
         onRequestClose();
```

Comment:
Please handle `onRequestClose` after the mutation is finished.
Also please set the  submit button's loading prop to mutation's `isPending`